tests/ludwig/utils/test_backward_compatibility.py

```diff
@@ -929,3 +929,21 @@ def test_type_removed_from_defaults_config():
 
     for feature_type in config_obj_2.get("defaults"):
         assert "type" not in config_obj_2["defaults"][feature_type]
+
+
+def test_trainer_effective_batch_size():
+    config = {
+        "input_features": [
+            {"name": "Age", "type": "number"},
+        ],
+        "output_features": [
+            {"name": "Survived", "type": "category"},
+        ],
+        "trainer": {
+            "batch_size": 64,
+        },
+    }
+
+    config_obj = ModelConfig.from_dict(config)
+    assert config_obj.trainer.effective_batch_size
```

Comment:
Is this intended to check whether or not the attribute exists? If so, it might be a good idea to refactor to

```
assert hasattr(config_obj.trainer, "effective_batch_size")
```

This would be a bit clearer and avoids any collisions with `effective_batch_size` evaluating to `False` (admittedly also an issue, but we should be able to catch those with the second assert).